### DISCUSSION TITLE
用条件配置添加滤镜的时候，是不是应该避免使用'-toggle'

### DISCUSSION THREAD

**Discussion by yeezylife** [dyphire/mpv-config#23] [General]
感谢你平时回答的我的问题，之前你教我写过一个设置：
[audio-filter]
profile-desc=音频通道大于2时开启动态调节滤镜，适用于双通道设备
profile-cond=p["audio-params/channel-count"] > 2
profile-restore=copy-equal
af-toggle=@dynaudnorm:lavfi=[dynaudnorm=g=5:f=250:r=0.9:p=0.5]

就拿这个设置打比方，我发现mpv的watch later功能（有时候，比如我手动开启/关闭过一次这个滤镜的时候）会记住这个滤镜（@dynaudnorm:lavfi=[dynaudnorm=g=5:f=250:r=0.9:p=0.5]）

这种情况一旦出现，下次我打开同一个文件的时候，条件配置启用af-toggle反而会去掉这个滤镜了。

我目前自己把条件配置文件下的af-toggle改成了af-set，这样mpv就算记住了同一个滤镜，下次打开也一定会按照af-set的滤镜加载。

我认为条件配置的自动处理，应该是”绝对“且让人”放心“的，用-set的话，无论mpv有没有”记住“你的设置，下次打开一定会被-set的设置覆盖掉，这样一定不会出现自己设置的之外的情况

**Comment by dyphire** [comment]
嗯，抓的好。通过条件配置添加滤镜时是该避免使用'-toggle'，已在最新提交中修正：https://github.com/dyphire/mpv-config/commit/c3be03cb8178b783fe0d9ff01030de4c41e5daa3
